### DISCUSSION TITLE
No matched pair in pointstat

### DISCUSSION THREAD

**Discussion by ghost** [dtcenter/METplus#2428] [Configuration] [answered]
I am trying to match forecast and observation using pointstat but getting no matching pairs. 
DEBUG 2: Number of matched pairs   = 0
DEBUG 2: Observations processed    = 859766
DEBUG 2: Rejected: station id      = 0
DEBUG 2: Rejected: obs var name    = 476005
DEBUG 2: Rejected: valid time      = 379511
DEBUG 2: Rejected: bad obs value   = 0
DEBUG 2: Rejected: off the grid    = 1950
DEBUG 2: Rejected: topography      = 0
DEBUG 2: Rejected: level mismatch  = 2208
DEBUG 2: Rejected: quality marker  = 0
DEBUG 2: Rejected: message type    = 92

Location on Hera /scratch1/NCEPDEV/stmp2/Partha.Bhattacharjee/METplus/Scripts/Regional/run_metplus_Pointstat_Regn.sh
logfile /scratch1/NCEPDEV/stmp2/Partha.Bhattacharjee/METplus/Scripts/Regional/logs

Any help on this appreciated. 
Thanks

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your submission. Reviewing the debug messages you provided here, it looks like the biggest rejection issues are `obs var name` and `valid time`. `obs var name` is often the largest source of rejected matched pairs due to the structure of the observation file and how many additional variables are included, so this seems like an acceptable level (~50% of observations). `valid time` is also a normally high level of rejections, but can be controlled with `OBS_POINT_STAT_WINDOW_BEGIN` and `OBS_POINT_STAT_WINDOW_END`. I saw in your configuration file you have both set to +/- 30 minutes. If this is your desired window then the rejection level seems appropriate.

The biggest things to check are the last two rejections that have values in the list. `message type` is a common culprit of matched pairs that should be included, but are tossed out due to incorrect settings of the `POINT_STAT_MESSAGE_TYPE`. When I used the print_pointnc2ascii.py script ([included in the MET installation](https://met.readthedocs.io/en/latest/Users_Guide/reformat_point.html#point-netcdf-to-ascii-python-utility)) to take a look at the point observations in your All_20220508.lev15.nc observation file, I found "AERONET_AOD" listed as the message type. However, this does NOT match what you provided in the configuration file:
```
POINT_STAT_MESSAGE_TYPE = AERONET
```
As a result you will not get any observations matching your message type. Adjust your configuration file to 
```
POINT_STAT_MESSAGE_TYPE = AERONET_AOD
```
and those previously 92 rejected observations should be able to be properly sorted. If you continue to run into problems, feel able to reach out for additional guidance.

**Comment by j-opatz** [comment]
Hello,

It's been some time since there was any discussion on this question. Did the suggested actions solve your problem? If so, please mark a comment as the answer so this question can be closed.
